src/appleseed/foundation/math/fresnel.h

```diff
@@ -435,50 +448,70 @@ void normal_reflectance_dielectric(
 template <typename SpectrumType, typename T>
 inline void fresnel_reflectance_conductor(
     SpectrumType&           reflectance,
-    const SpectrumType&     n,
-    const SpectrumType&     k,
+    const SpectrumType&     nt,
+    const SpectrumType&     kt,
+    const T                 ni,
     const T                 cos_theta_i)
 {
-    //
-    //      (n^2 + k^2) cos_theta^2 - 2n cos_theta + 1
-    // Rp = ------------------------------------------
-    //      (n^2 + k^2) cos_theta^2 + 2n cos_theta + 1
-    //
+    const T cos_theta = std::abs(cos_theta_i);
+    const T cos_theta2 = square(cos_theta);
+    const T sin_theta2 = T(1.0) - cos_theta2;
+    const T sin_theta4 = square(sin_theta2);
 
-    //
-    //      (n^2 + k^2) - 2n cos_theta + cos_theta^2
-    // Rs = ----------------------------------------
-    //      (n^2 + k^2) + 2n cos_theta + cos_theta^2
-    //
+    const T rcp_ni2 = T(1.0) / square(ni);
 
-    //               Rp + Rs
-    // reflectance = -------
-    //                  2
+    SpectrumType nt2 = nt * nt;
+    nt2 *= rcp_ni2;
 
-    typedef typename impl::GetValueType<SpectrumType>::ValueType ValueType;
+    SpectrumType kt2 = kt * kt;
+    kt2 *= rcp_ni2;
 
-    const T cos_theta = std::abs(cos_theta_i);
-    const T cos_theta2 = square(cos_theta);
+    SpectrumType t0 = nt2 - kt2;
+    t0 -= SpectrumType(sin_theta2);
 
-    const SpectrumType n2 = n * n;
-    const SpectrumType k2 = k * k;
-    const SpectrumType n2_plus_k2 = n2 + k2;
-    const SpectrumType n_2cos_theta = n * ValueType(2.0) * static_cast<ValueType>(cos_theta);
+    SpectrumType a2plusb2 = t0 * t0;
+    SpectrumType tmp = nt2 * kt2;
+    tmp *= T(4.0);
+    a2plusb2 += tmp;
+    a2plusb2 = sqrt(a2plusb2);
```

Comment:
Yes, but it would be incorrect to call this method with SpectrumType = float or double in any case..